test/e2e/storage/csi_mock_volume.go

```diff
@@ -298,6 +299,62 @@ var _ = utils.SIGDescribe("CSI mock volume", func() {
 		}
 	})
 
+	ginkgo.Context("CSI CSIDriver deployment after pod creation using non-attachable mock driver", func() {
+		ginkgo.It("should bringup pod after deploying CSIDriver attach=false [Slow]", func() {
+			var err error
+			init(testParameters{registerDriver: false, disableAttach: true})
+			defer cleanup()
+
+			_, claim, pod := createPod(false)
+			if pod == nil {
+				return
+			}
+			err = e2epod.WaitTimeoutForPodRunningInNamespace(m.cs, pod.Name, pod.Namespace, 10*time.Second)
```

Comment:
I think 10 seconds might be too small value for pod to fail to start, I know this will add to the test suite time but how about something like 1 minute?

Reply:
Most of our e2es are using the podstartuptimeout defined in the e2e framework, which is 5 minutes I believe

Reply:
yeah - I kind of hate to add definitive 5 minute delay to the test suite. In other places 5 minute is the worst case timeout but in this case it will for sure hang around for 5minutes before erroring out.

Reply:
We can't rely on events, but what we've done in other negative test cases is to exit the condition early if we find the event we're looking for, otherwise wait for the full 5 minutes

Reply:
@msau42 are you suggesting we should check events while waiting here? If so, can you give me an example of the negative test cases you mentioned?

Reply:
https://github.com/kubernetes/kubernetes/blob/6d0f4749a59099171540d4fd7c9523b029e71ceb/test/e2e/storage/testsuites/volumemode.go#L239

Reply:
or if there's a pod status/condition we can check to speed up the wait

Reply:
Done. I end up checking the events, dont know if there is a status or condition we can check here. An event seems to be working fine.